WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+FAB.swift

```diff
@@ -33,11 +35,52 @@ extension MySiteViewController {
         }
 
         actions.append(PostAction(handler: newPost, source: source))
+        // TODO: check if the current site is eligible
+        if FeatureFlag.voiceToContent.enabled {
+            actions.append(PostFromAudioAction(handler: { [weak self] in
+                self?.dismiss(animated: true) {
+                    self?.startPostFromAudioFlow()
+                }
+            }, source: source))
+        }
         if blog?.supports(.pages) ?? false {
             actions.append(PageAction(handler: newPage, source: source))
         }
 
         let coordinator = CreateButtonCoordinator(self, actions: actions, source: source, blog: blog)
         return coordinator
     }
+
+    private func startPostFromAudioFlow() {
+        guard let blog else {
+            wpAssertionFailure("blog missing")
+            return
+        }
+        let viewModel = VoiceToContentViewModel(blog: blog) { [ weak self] transcription in
+            guard let self else { return }
+            self.dismiss(animated: true) {
+                // TODO: Are we adding all necessary fields?
+                let presenter = RootViewCoordinator.sharedPresenter
+                let post = blog.createDraftPost()
+                post.content = """
+                <!-- wp:paragraph -->
+                <p>\(transcription.escapeHtmlNamedEntities())</p>
+                <!-- /wp:paragraph -->
+                """
+                presenter.showPostTab(animated: true, post: post)
+            }
+        }
+        let view = VoiceToContentView(viewModel: viewModel)
+        let host = UIHostingController(rootView: view)
+
+        if UIDevice.isPad() {
+            host.modalPresentationStyle = .formSheet
+        } else {
+            if let sheetController = host.sheetPresentationController {
+                sheetController.detents = [.medium()]
```

Comment:
On smaller devices, "Post from audio" is obscured in the sheet

iPhone SE (17.4.1) | iPhone 15 Pro (17.0) 
-- | --
<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/1444adde-0cff-45b1-9807-c756f236315b" width=200> | <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/f342e228-9ae8-4eb7-9f4c-b9f41192ceac" width=200>

Reply:
I suggest addressing it when we start working on the final design.